src/validators/validator.ts

```diff
@@ -28,85 +17,50 @@ export class Validator {
     return Array.from(this.errors).join('\r\n') || null;
   }
 
-  public hasLowerCase(): Validator {
-    this.validateSync(new HasLowerCase());
-    return this;
-  }
-
-  public hasUpperCase(): Validator {
-    this.validateSync(new HasUpperCase());
-    return this;
-  }
-
   public isConfirmationValue(): Validator {
```

Comment:
couldn't this and the other functions in here be simplified/chained better into arrow functions? 

If your purpose is to make them more functional then there is still a bunch of boilerplate you can remove here to streamline the code and get the same values.

Reply:
Yes, all this has been removed. I replaced subclassing with arrow functions. Is that what you meant?

Reply:
but cant the outer function that returns the function be arrowed too?

YES_NO can be either turned into a const up above (otherwise you redefine it every call unncessarily) or inline it into the map call?

you're really just returning curried functions at this point, it would look clearer to me to return the partial/completed function as a function v's the return syntax inside a function.

Reply:
> you're really just returning curried functions at this point

`validate` returns `this` to be able to chain methods. Not sure how would you accomplish the same using curryfication.

Reply:
> YES_NO can be either turned into a const up above (otherwise you redefine it every call unncessarily) or inline it into the map call?

Sounds good, will move it to an outer scope.

Reply:
>> YES_NO can be either turned into a const up above (otherwise you redefine it every call unncessarily) or inline it into the map call?

> Sounds good, will move it to an outer scope.

After a second though, I usually prefer to use the smallest scope when redefinition is negligible.

Reply:
Hey @michaeljohnbennett, do you think that we can merge this? Or you are strongly inclined to refactor it? Just asking so I can move forward with another PR without conflicts.

Reply:
let me see what you ended up doing. Generally I just like to collapse or make apis as chainable as possible without doing a whole bunch of internal code where you could obviate it. 

I'm sure it will be fine to proceed. It's easy enough to enhance later on if we decide to.